pkg/controller/supportbundlecollection/validate.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package supportbundlecollection
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+
+	admv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+
+	crdv1alpha1 "antrea.io/antrea/pkg/apis/crd/v1alpha1"
+)
+
+func (c *Controller) Validate(review *admv1.AdmissionReview) *admv1.AdmissionResponse {
+	klog.V(2).Info("Validating SupportBundleCollection", "request", review.Request)
+	var newObj, oldObj crdv1alpha1.SupportBundleCollection
+	if review.Request.Object.Raw != nil {
+		if err := json.Unmarshal(review.Request.Object.Raw, &newObj); err != nil {
+			klog.ErrorS(err, "Error de-serializing current SupportBundleCollection")
+			return newAdmissionResponseForErr(err)
+		}
+	}
+	if review.Request.OldObject.Raw != nil {
+		if err := json.Unmarshal(review.Request.OldObject.Raw, &oldObj); err != nil {
+			klog.ErrorS(err, "Error de-serializing old IPPool")
+			return newAdmissionResponseForErr(err)
+		}
+	}
+
+	validateProcessingCollection := func(checkSpecUpdateOnly bool) *admv1.AdmissionResponse {
+		var msg string
+		allowed := true
+		_, exists, _ := c.supportBundleCollectionStore.Get(oldObj.Name)
+		if exists {
+			if checkSpecUpdateOnly {
+				allowed = reflect.DeepEqual(oldObj.Spec, newObj.Spec)
+			} else {
+				allowed = false
+			}
+			if !allowed {
+				msg = fmt.Sprintf("SupportBundleCollection %s is started, cannot be updated or deleted", oldObj.Name)
```

Comment:
why it doesn't allow deleting a started collection? technical reason?

Reply:
According to the offline discussion, I would remove the validation in deletion even if the support bundle collection is already started after confirmed the impact on agent logic.

Reply:
should the message be updated? can a started request be deleted? 
How do we know it has started here?

Reply:
updated the message, a started request can be deleted. The "started" condition is added in another PR for status update.